website/docs/r/nsxt_alb_virtual_service_http_req_rules.html.markdown

```diff
@@ -0,0 +1,197 @@
+* `rule` - (Required) One or more [rule](#rule-block) blocks with matching criteria and actions.
+
+<a id="rule-block"></a>
+## Rule
+
+* `name` - (Required) Name of the rule
+* `active` - (Optional) Defines if the rule is active. Default `true`
+* `logging` - (Optional) Defines if the requests that match should be logged. Default `false`
+* `match_criteria` - (Required) A block of [criteria](#rule-criteria-block) to match the requests
+* `actions` - (Required) A block of [actions](#rule-action-block) to perform with requests that match
+
+<a id="rule-criteria-block"></a>
+## Match Criteria
+
+One or more of criteria can be specified to match traffic. At least one criteria is required
+
+* `client_ip_address` - (Optional) Match the rule based on client IP address rules
+ * `criteria` - (Required) One of `IS_IN`, `IS_NOT_IN`
+ * `ip_addresses` - (Required) A set of IP addresses to match
+* `service_ports` - (Optional) Match the rule based on service ports
+ * `criteria` - (Required) One of `IS_IN`, `IS_NOT_IN`
+ * `ports` - (Required) A set of ports to match
+* `protocol_type` - (Optional) One of `HTTP` or `HTTPS`
+* `http_methods` - (Optional) Defines HTTP methods that should be matched
+ * `criteria` - (Required) One of `IS_IN`, `IS_NOT_IN`
+ * `methods` - (Required) A set of HTTP methods from the list: `GET`, `PUT`, `POST`, `DELETE`,
+   `HEAD`, `OPTIONS`, `TRACE`, `CONNECT`, `PATCH`, `PROPFIND`, `PROPPATCH`, `MKCOL`, `COPY`, `MOVE`,
+   `LOCK`, `UNLOCK`
+* `path` - (Optional) 
+ * `criteria` - (Required) One of `BEGINS_WITH`, `DOES_NOT_BEGIN_WITH`, `CONTAINS`, `DOES_NOT_CONTAIN`, `ENDS_WITH`, `DOES_NOT_END_WITH`, `EQUALS`, `DOES_NOT_EQUAL`, `REGEX_MATCH`, `REGEX_DOES_NOT_MATCH`
+ * `paths` - (Required) A set of patchs to match given criteria
+* `query` - (Optional) HTTP request query strings to match
+* `request_headers` - (Optional) One or more blocks of request headers to match
+ * `criteria` - (Required) One of `BEGINS_WITH`, `DOES_NOT_BEGIN_WITH`, `CONTAINS`, `DOES_NOT_CONTAIN`, `ENDS_WITH`, `DOES_NOT_END_WITH`, `EQUALS`, `DOES_NOT_EQUAL`, `REGEX_MATCH`, `REGEX_DOES_NOT_MATCH`
+ * `name` - (Required) Name of the header to match
+ * `values` - (Required) A set of strings values that should match header value
+* `cookie` - (Optional) A block 
+ * `criteria` - (Required) One of `BEGINS_WITH`, `DOES_NOT_BEGIN_WITH`, `CONTAINS`, `DOES_NOT_CONTAIN`, `ENDS_WITH`, `DOES_NOT_END_WITH`, `EQUALS`, `DOES_NOT_EQUAL`, `REGEX_MATCH`, `REGEX_DOES_NOT_MATCH`
+ * `name` - (Required) Name of the HTTP cookie to match
+ * `value` - (Required) A cookie value to match
+
+<a id="rule-action-block"></a>
+## Actions
+
+One or more of action blocks can be specified. At least one is required. Some may prohibit others
+and API validation will return errors
+
+* `redirect` - (Optional) Redirects the request to different location
+ * `protocol` - (Required) One of `HTTP`, `HTTPS`
+ * `port` - (Required) Destination port for redirect
```

Comment:
Should we mention "Default is 80 for HTTP and 443 for HTTPS protocol"?

Reply:
Can do. Changed.